lib/rubocop/cop/minitest/assert_equal_arguments_order.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # This cop enforces correct order of expected and
+      # actual arguments for `assert_equal`.
+      #
+      # @example
+      #   # bad
+      #   assert_equal foo, 2
+      #   assert_equal foo, [1, 2]
+      #   assert_equal foo, [1, 2], 'message'
+      #
+      #   # good
+      #   assert_equal 2, foo
+      #   assert_equal [1, 2], foo
+      #   assert_equal [1, 2], foo, 'message'
+      #
+      class AssertEqualArgumentsOrder < Cop
+        MSG = 'You misordered expected and actual arguments for `assert_equal`.'
```

Comment:
For example, the following message may be better.

```suggestion
        MSG = 'Replace the literal with the first argument.'
```